pygmt/clib/session.py

```diff
@@ -1067,6 +1067,98 @@ def put_matrix(self, dataset, matrix, pad=0):
         if status != 0:
             raise GMTCLibError(f"Failed to put matrix of type {matrix.dtype}.")
 
+    def read_data(
+        self,
+        infile: str,
+        kind: Literal["dataset", "grid"],
+        family: str | None = None,
+        geometry: str | None = None,
+        mode: str = "GMT_READ_NORMAL",
```

Comment:
Thinking on whether to make this an enum or use `Literal` instead of a str. Looking at https://docs.generic-mapping-tools.org/6.5/devdocs/api.html#import-from-a-file-stream-or-handle, it seems like it's possible to pass in `mode` as one of the variants of [`GMT_enum_container`](https://github.com/GenericMappingTools/gmt/blob/3e9fcb1f54e4e93fc1e309e0be6e2288eb79f3d2/src/gmt_resources.h#L136-L143) or [`GMT_enum_gridio`](https://github.com/GenericMappingTools/gmt/blob/3e9fcb1f54e4e93fc1e309e0be6e2288eb79f3d2/src/gmt_resources.h#L363-L379)? It also looks like `mode` can be None (or unset) if reading from text or table.

Reply:
I believe the documentation is incomplete, and `mode` can accept specific values plus some modifiers, making it impossible to list all of them.

Reply:
Yeah that what I thought too, there might be other options we's unaware of. That said, should we allow for `mode=None` (or whatever works when we don't want to set a `mode`)?

Reply:
In the GMT C API, `mode` has an `unsigned int` type, so the `mode` argument must be an integer number and can't be `None`. The C enum `GMT_READ_NORMAL` has a value of `0` (https://github.com/GenericMappingTools/gmt/blob/3e9fcb1f54e4e93fc1e309e0be6e2288eb79f3d2/src/gmt_resources.h#L244), which is also the default read mode.

We can define `mode=None`, and add one line code like:
```
if mode == None:
   mode = "GMT_READ_NORMAL"
```
but I feel it's not as good as setting `GMT_READ_NORMAL` as the default.

Reply:
Ah ok then, then the `GMT_READ_NORMAL` default here makes sense. Maybe good to mention it in the docstring.